### ISSUE TITLE
Refinery panics on rules reload

### ISSUE THREAD

**Issue by irvingpop** [honeycombio/refinery#595]
More of an annoyance than anything, but I noticed Refinery sending this to the logs whenever it live-reloads the rules:

```
panic: send on closed channel
goroutine 30414 [running]:
github.com/honeycombio/libhoney-go/transmission.(*Honeycomb).tryAdd(0x4012a63340, 0x4000248bd0)
        /home/circleci/go/pkg/mod/github.com/honeycombio/libhoney-go@v1.18.0/transmission/transmission.go:231 +0x224
github.com/honeycombio/libhoney-go/transmission.(*Honeycomb).Add(0x4012a63340, 0x4000248bd0)
        /home/circleci/go/pkg/mod/github.com/honeycombio/libhoney-go@v1.18.0/transmission/transmission.go:204 +0x28
github.com/honeycombio/libhoney-go.(*Event).SendPresampled(0x4029b57040)
        /home/circleci/go/pkg/mod/github.com/honeycombio/libhoney-go@v1.18.0/libhoney.go:885 +0x57c
github.com/honeycombio/libhoney-go.(*Event).Send(0x4029b57040)
        /home/circleci/go/pkg/mod/github.com/honeycombio/libhoney-go@v1.18.0/libhoney.go:811 +0x9c
github.com/honeycombio/refinery/metrics.(*HoneycombMetrics).reportToHoneycomb(0x40000c8800, {0x9eda10, 0x4007b4dd00})
        /home/circleci/project/metrics/honeycomb.go:265 +0x58
created by github.com/honeycombio/refinery/metrics.(*HoneycombMetrics).initLibhoney
        /home/circleci/project/metrics/honeycomb.go:143 +0x4a4
refinery.service: Main process exited, code=exited, status=2/INVALIDARGUMENT
refinery.service: Failed with result 'exit-code'.
refinery.service: Consumed 2min 37.012s CPU time.
```

**Versions**

- Go:  1.19.3
- Refinery: 1.19.0 (also observed with 1.18)

**Steps to reproduce**

1.  Configure Refinery to have metrics and logs sent to Honeycomb (config snippet below)
2. Update the rules file on disk
3. watch the logs  ( `journalctl -f -u refinery.service`)
4. Seems to happen every time as far as I can tell

relevant config
```
Logger = "honeycomb"
Metrics = "honeycomb"

[HoneycombLogger]
LoggerHoneycombAPI = "https://api.honeycomb.io"
LoggerAPIKey = "${HONEYCOMB_API_KEY}"
LoggerDataset = "Refinery Logs"
LoggerSamplerEnabled = true
LoggerSamplerThroughput = 10

[HoneycombMetrics]
MetricsHoneycombAPI = "https://api.honeycomb.io"
MetricsAPIKey = "${HONEYCOMB_API_KEY}"
MetricsDataset = "Refinery Metrics"
MetricsReportingInterval = 60
```

**Additional context**
Not sending refinery process any signals, all that happens just when the config file is updated.

**Comment by kentquirk** [comment]
Thanks! I know roughly what's going on, but it's affected by some of the in-flight work, so I will land those and come back to this one soon.

**Comment by kentquirk** [comment]
Hi, @irvingpop -- I'm having trouble reproducing this in the current head of refinery. I don't know if it's a rules-file specific thing, something I fixed accidentally when reorganizing metrics, or if there's some other factor that influences it.

Could you please include the rules file you're using too, just to see if that has an effect?

**Comment by irvingpop** [comment]
Sure!   FWIW I got this result running the `refinery-1.19.0-1.aarch64` RPM, on Amazon Linux.

refinery.toml:
```
ListenAddr = "0.0.0.0:8080"
GRPCListenAddr = "0.0.0.0:9090"
PeerListenAddr = "0.0.0.0:8081"
CompressPeerCommunication = true
APIKeys = [
  "*",                   # wildcard accept all keys
  ]
HoneycombAPI = "https://api.honeycomb.io"
SendDelay = "2s"
TraceTimeout = "60s"
MaxBatchSize = 500
SendTicker = "100ms"
LoggingLevel = "info"
UpstreamBufferSize = 10000
PeerBufferSize = 10000
AddHostMetadataToTrace = true
QueryAuthToken = "****"
AddRuleReasonToTrace = true
AddSpanCountToRoot = true
CacheOverrunStrategy = "impact"
Collector = "InMemCollector"
Logger = "honeycomb"
Metrics = "honeycomb"

[PeerManagement]
Type = "redis"
RedisHost = "redacted:6379"
RedisPassword = "****"
UseTLS = true

[InMemCollector]
CacheCapacity = 319291
MaxAlloc = 3192913920

[HoneycombLogger]
LoggerHoneycombAPI = "https://api.honeycomb.io"
LoggerAPIKey = "****"
LoggerDataset = "Refinery Logs"
LoggerSamplerEnabled = true
LoggerSamplerThroughput = 10

[HoneycombMetrics]
MetricsHoneycombAPI = "https://api.honeycomb.io"
MetricsAPIKey = "****"
MetricsDataset = "Refinery Metrics"
MetricsReportingInterval = 60
```

rules.toml:
```
LiveReload = true
SampleRate = 1
Sampler = "DeterministicSampler"

[prod]
    Sampler = "EMADynamicSampler"
    GoalSampleRate = 50
    FieldList = ["status_code", "http.url", "service.name", "name"]
    UseTraceLength = true
    AddSampleRateKeyToTrace = true
    AddSampleRateKeyToTraceField = "meta.refinery.dynsampler_key"
    AdjustmentInterval = 60
    MaxKeys = 10000
    Weight = 0.5
```

**Comment by bixu** [comment]
We've seen an issue similar to this, where a Rules Config Map change in K8s causes Refinery to fail to pass traces upstream. In our case, we'd see the `/alive` route return 200, but no traces make it to Honeycomb.

cc @BouchaaraAdil / @www44

**Comment by TylerHelmuth** [comment]
We were able to reproduce this issue in 2.0.2 by enabling LiveReload and modifying the rulesconfigmap.  We are continuing to investigate root cause.

**Comment by TylerHelmuth** [comment]
Turns out we fixed this: https://github.com/honeycombio/refinery/issues/807

**Comment by kentquirk** [comment]
It was reported in #807, fixed in #810.
